### ISSUE TITLE
Decouple Alsa device buffer and period sizes from CamillaDSP chunksize

### ISSUE THREAD

**Issue by HEnquist** [HEnquist/camilladsp#376]
The Wasapi and CodeAudio capture and playback devices use a an outer and an inner thread, with a buffer in between. This allows the inner thread to read or write as many frames as the device requires, while the outer threads exchange chunks with _chunksize_ frames with the processing pipeline.
The Alsa backend uses a single thread per device, which means the device buffer and period sizes cannot be chosen freely. This is mostly ok since most devices support wide ranges for both parameters. But some devices do not, for example requiring the buffer size to be 2 * period size.
Rewrite the Alsa backend to add an inner like in the Wasapi and CodeAudio backends to get around this limitation.

**Comment by siraaris** [comment]
Would this feature support/enable setting the Loopback timer_source (Camilla capture dev) to the DAC (Camilla playback dev)?

**Comment by HEnquist** [comment]
Yes this should allow the playback device to start even if no chunk has been received yet. In the wasapi and coreaudio backends the playback device just plays silence when there is no data from the pipeline, and it will most likely end up like that for Alsa as well.

**Comment by siraaris** [comment]
I had some fun with OpenAI/Codex and got this working, based on CoreAudio and Wasapi implementations.

I suspect you may have other considerations for the ALSA backend, but for interest here is the code - https://github.com/siraaris/camilladsp there's only one branch.

**Comment by HEnquist** [comment]
There is an alternative Alsa backend included since the 4.1.0 release. Compile with the t "threaded-alsa" feature enabled to switch to this new backend. If you try it, please report back how it performs.

**Comment by siraaris** [comment]
I'll have a look, thanks!

**Comment by siraaris** [comment]
Running it now, will let you know. I found some oddities with teardown (see #459).
